samcli/lib/providers/sam_function_provider.py

```diff
@@ -309,13 +343,19 @@ def _parse_layer_info(list_of_layers, resources, ignore_code_extraction_warnings
 
             # If the layer is a string, assume it is the arn
             if isinstance(layer, str):
-                layers.append(LayerVersion(layer, None))
+                layers.append(
+                    LayerVersion(
+                        layer,
+                        None,
+                        stack_path=stack_path,
+                    )
+                )
                 continue
 
             # In the list of layers that is defined within a template, you can reference a LayerVersion resource.
             # When running locally, we need to follow that Ref so we can extract the local path to the layer code.
             if isinstance(layer, dict) and layer.get("Ref"):
-                layer_logical_id = layer.get("Ref")
+                layer_logical_id = cast(str, layer.get("Ref"))
```

Comment:
Curious, why did we added casting here?

Reply:
`layer.get("Ref")` return `Any`, since we know it is already `str` I do the cast here.